pages/tutorials/3.0/window/events.md

```diff
@@ -7,259 +7,329 @@
+However, it is triggered only when the mouse moves within the inner area of the window, not when it moves over the title bar or borders.
+
+The data associated with this event contains the current position of the mouse cursor relative to the window.
+
+```cpp
+if (const auto* mouseMoved = event->getIf<sf::Event::MouseMoved>())
+{
+    std::cout << "new mouse x: " << mouseMoved->position.x << std::endl;
+    std::cout << "new mouse y: " << mouseMoved->position.y << std::endl;
+}
+```
+
+## The MouseMovedRaw event
+
+The `sf::Event::MouseMovedRaw` event is triggered when the mouse moved within the window even if the mouse is moved a distance too small to perceive.
 
-This event is triggered even if the window isn't focused. However, it is triggered only when the mouse moves within the inner area of the window, not when it moves over the title bar or borders.
+While the MouseMoved position value is dependent on the screen resolution, raw data is not.
+If the physical mouse is moved too little to cause the screen cursor to move at least a single pixel, no MouseMoved event will be generated.
+In contrast, any movement information generated by the mouse independent of its sensor resolution will always generate a `MouseMovedRaw` event.
 
-The member associated with this event is `event.mouseMove`, it contains the current position of the mouse cursor relative to the window.
+The data associated with this event contains the change in position of the mouse cursor relative to the window.
 
 ```cpp
-if (event.type == sf::Event::MouseMoved)
+if (const auto* mouseMovedRaw = event->getIf<sf::Event::MouseMovedRaw>())
 {
-    std::cout << "new mouse x: " << event.mouseMove.x << std::endl;
-    std::cout << "new mouse y: " << event.mouseMove.y << std::endl;
+    std::cout << "new mouse x: " << mouseMoved->delta.x << std::endl;
+    std::cout << "new mouse y: " << mouseMoved->delta.y << std::endl;
 }
 ```
 
+Currently, raw mouse input events will only be generated on Windows and Linux.
+
 ## The MouseEntered and MouseLeft event
 
 The `sf::Event::MouseEntered` and `sf::Event::MouseLeft` events are triggered when the mouse cursor enters/leaves the window.
-
-There's no member associated with these events in the [`sf::Event`](https://www.sfml-dev.org/documentation/3.0.0/classsf_1_1Event.php "sf::Event documentation") union.
+There's no data associated with these events.
 
 ```cpp
-if (event.type == sf::Event::MouseEntered)
+if (event->is<sf::Event::MouseEntered>())
     std::cout << "the mouse cursor has entered the window" << std::endl;
 
-if (event.type == sf::Event::MouseLeft)
+if (event->is<sf::Event::MouseLeft>())
     std::cout << "the mouse cursor has left the window" << std::endl;
 ```
 
 ## The JoystickButtonPressed and JoystickButtonReleased events
 
 The `sf::Event::JoystickButtonPressed` and `sf::Event::JoystickButtonReleased` events are triggered when a joystick button is pressed/released.
-
 SFML supports up to 8 joysticks and 32 buttons.
```

Comment:
I think this makes sense as a new paragraph.

Reply:
It felt a bit awkward to have 3 sentences in a row all on their own paragraphs. Felt like we were adding whitespace for no good reason. I'll restore that newline though.